gatsby-site/src/components/Newsletter.js

```diff
@@ -0,0 +1,104 @@
+
+const LinkContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  padding-top: 20px;
+`
+
+const ActionNetwork = ({ actionId }) => {
+  const container = React.useRef()
+
+  const onInput = React.useCallback(event => {
+    const target = event.target
+    target.value === ""
+      ? target.parentElement.classList.remove("has-value")
+      : target.parentElement.classList.add("has-value")
+    target.classList.contains("error_input")
+      ? target.parentElement.classList.add("has-error")
+      : target.parentElement.classList.remove("has-error")
+  }, [])
+
+  const onBlur = React.useCallback(event => {
+    const target = event.target
+    target.classList.contains("error_input")
+      ? target.parentElement.classList.add("has-error")
+      : target.parentElement.classList.remove("has-error")
+  }, [])
+
+  React.useEffect(() => {
+    if (!container.current) return
+    let inner
+    if (actionNetworkCache.has(actionId)) {
+      inner = actionNetworkCache.get(actionId)
+      container.current.appendChild(inner)
+    } else {
+      inner = document.createElement("div")
+      inner.innerHTML = `
+        <div id="can-form-area-${actionId}" style="width: 100%;"></div>
+      `
+      const script = document.createElement("script")
+      script.src = `https://actionnetwork.org/widgets/v3/form/${actionId}?format=js&source=widget`
+      inner.appendChild(script)
+      const styles = document.createElement("link")
+      styles.href =
+        "https://actionnetwork.org/css/style-embed-whitelabel-v3.css"
+      styles.rel = "stylesheet"
+      inner.appendChild(styles)
+      container.current.appendChild(inner)
+    }
+    inner.addEventListener("input", onInput)
+    inner.addEventListener("focusout", onBlur)
+    return () => {
+      inner.removeEventListener("input", onInput)
+      inner.removeEventListener("focusout", onBlur)
+      actionNetworkCache.set(actionId, inner)
+      container.current && container.current.removeChild(inner)
+    }
+  }, [actionId])
+
+  return <div className={Styles.container} ref={container} />
+}
+
+export default ({ link }) => {
+  const url = React.useMemo(() => {
+    return new URL(link)
+  }, [link])
+
+  const actionNetworkActionId = React.useMemo(() => {
+    if (url.hostname !== "actionnetwork.org") return null
+    const [type, actionId] = url.pathname.split("/").filter(x => x)
+    console.log(type)
+    if (type !== "forms" || !actionId) return null
+    return actionId
+  }, [url])
+
+  const apexDomain = React.useMemo(() => {
+    return url.host
+      .split(".")
+      .slice(-2)
+      .join(".")
+  }, [url])
+
+  if (actionNetworkActionId) {
+    return (
+      <Section id="newsletter" title="Newsletter Sign Up">
+        <ActionNetwork actionId={actionNetworkActionId} />
+      </Section>
+    )
+  }
+
+  return (
+    <Section id="newsletter" title="Newsletter Sign Up">
+      <LinkContainer>
+        <Button label={`Sign up at ${apexDomain}`} href={link} />
```

Comment:
Just from scanning some of the [eventual production data](https://sunrise-hub-json.s3.amazonaws.com/hubs.json), it looks like some of the apex domains will be things like `bit.ly` and `forms.gle`. That might be fine but it might also look a bit weird, so just wanted to flag it!